src/centralserver/admin-service/src/main/resources/openapi-definition.yaml

```diff
@@ -1,357 +1,4105 @@
+          type: string
+        instance_id:
+          description: Xroad instance id. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: FI
+          format: text
+          maxLength: 255
+          minLength: 1
+          readOnly: true
+          type: string
+        member_type:
+          $ref: '#/components/schemas/MemberType'
+        subsystem_code:
+          description: Subsystem code. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: ABC
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+      type: object
+    MemberClass:
+      allOf:
+        - $ref: '#/components/schemas/MemberClassDescription'
+      description: Central server's member classes
+      properties:
+        code:
+          description: member class code
+          example: ORG
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+      required:
+        - code
+      type: object
+    MemberClassAndCode:
+      description: x-road member class and code
+      properties:
+        member_class:
+          description: Member class. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: GOV
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+        member_code:
+          description: Member code. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: 123
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+      required:
+        - member_class
+        - member_code
+      type: object
+    MemberClassDescription:
+      description: Central server's member class description
+      properties:
+        description:
+          description: member class description
+          example: Non-profit organisations
+          format: text
+          maxLength: 255
+          minLength: 1
+          type: string
+      required:
+        - description
+      type: object
+    MemberName:
+      description: x-road member name
+      properties:
+        member_name:
+          description: Member name. Cannot contain colons, semicolons, slashes, backslashes, percent signs or control characters
+          example: FI
+          format: text
+          maxLength: 255
+          minLength: 1
+          readOnly: true
+          type: string
+      required:
+        - member_name
+      type: object
+    MemberNameClassAndCode:
+      allOf:
+        - $ref: '#/components/schemas/MemberName'
+        - $ref: '#/components/schemas/MemberClassAndCode'
+      description: x-road member name, class and code
+      required:
+        - member_name
+        - member_class
+        - member_code
+      type: object
+    MemberType:
```

Comment:
I don't think these are "member" types? Subsystem is not a member, global group is not a member. Are these `ServiceClientType`s?  `Member` object probably shouldn't have MemberType (it can only ever be MemberType=MEMBER) as a property? And e.g. `GET /members` should not return global groups or subsystems(?)

Reply:
The `member` in Central Server context seems to be the same as `client` in Security Server context. So this is basically almost a copy of the Client model in Security Server – but just called "Member". They all (member, subsystem, globalgroup etc.) are derivatives of `XRoadId` and mostly have the same properties.

Perhaps better naming would be a good idea in this case? Client? 🤷‍♂️

Reply:
Hmm, wait, so is `findMembers` going to return also globalgroups? I assume not?

Member is such a "strongly typed" word in X-Road https://www.x-tee.ee/docs/live/xroad/terms_x-road_docs.html - AFAIK it means exactly a member, even subsystem can't be called member. Perhaps the endpoint `/members` should also be renamed, even if it operates only with members and subsystems, not global groups.

In SS these are called "clients", in CS DB the name seems to be "security server clients" https://github.com/nordic-institute/X-Road/blob/develop/doc/DataModels/dm-cs_x-road_central_server_configuration_data_model.md#222-security_server_clients

Reply:
addressed in https://github.com/nordic-institute/X-Road/commit/169431c4daa1a6168ad93faa3ff24b6a239172a2